test/cli/cli.test.js

```diff
@@ -78,13 +78,22 @@ describe('CLI', () => {
     testBin('--sockPath /mysockPath')
       .then((output) => {
         expect(
-          output.stdout.includes('http://localhost&sockPath=/mysockPath')
+          output.stdout.includes('http://localhost:8080&sockPath=/mysockPath')
```

Comment:
Don't use the port number directly. You can use [this file](https://github.com/webpack/webpack-dev-server/blob/master/test/ports-map.js).

Reply:
@hiroppy One problem with using the `ports-map` is that this is intentionally testing the case where the CLI defaults to a port when not specified. In such test cases, maybe `findPort` should be mocked to return a specific port.

Reply:
I just tried to mock `findPort` and `defaultPort` with `jest.mock()`. However, they do not work because `testBin` spawns a child process that does not have jest's mocked modules in it.

I think that for the purposes of this test, it's not important that findPort finds any particular port as much as it's important that it finds a port at all. I think I'll change the tests to just verify that a port was found at all, by matching the output to a regex `/localhost:[0-9+]/`

Thoughts?

Reply:
I just pushed code that does what I suggested ^

Let me know if you know of a better way to do this.

Reply:
Seems fine to me. One thing I have been thinking to make sure tests like this do not attach to a port that later creates a collision is to move the hard [port mappings](https://github.com/webpack/webpack-dev-server/blob/05bdb0ce0738c1d014c949575d15e77b6132ed25/test/ports-map.js#L49) up by 10, so that 8080, 8081, 8082, etc. are guaranteed available for these variable port tests. Not needed in this PR though, just an extra precaution.

Reply:
👍 I won't add that now but yeah I agree that would be cool in a future PR

Reply:
Yes, we need to reserve some ports for CLI tests.